solidity/test/bridge/Bridge.Redemption.test.ts

```diff
@@ -704,6 +704,232 @@ describe("Bridge - Redemption", () => {
     })
   })
 
+  describe("receiveBalanceApproval", () => {
+    const walletPubKeyHash = "0x8db50eb52063ea9d98b3eac91489a90f738986f6"
+    // Requested amount is 1901000 satoshi.
+    const requestedAmount = BigNumber.from(1901000)
+    // Treasury fee is `requestedAmount / redemptionTreasuryFeeDivisor`
+    // where the divisor is `2000` initially. So, we
+    // have 1901000 / 2000 = 950.5 though Solidity
+    // loses the decimal part.
+    const treasuryFee = 950
+
+    let redemptionTxMaxFee: BigNumber
+
+    let balanceOwner: SignerWithAddress
+    let redeemer: string
+
+    before(async () => {
+      redemptionTxMaxFee = (await bridge.redemptionParameters())
+        .redemptionTxMaxFee
+
+      balanceOwner = thirdParty
+      // eslint-disable-next-line prefer-destructuring
+      redeemer = (await getUnnamedAccounts())[10]
+
+      // Simulate the balance owner has a twice Bank balance allowing to make the request.
+      await bank.setBalance(balanceOwner.address, requestedAmount.mul(2))
+    })
```

Comment:
Looks like we modify the contract state here. It would be good to add a `createSnapshot` call before and an `after` section with `restoreSnapshot` to avoid strange failures in the future.

Reply:
Good catch. Fixed in https://github.com/keep-network/tbtc-v2/pull/296/commits/5da51b05bcf10d39f0835dbd28f7cae3337b8518.